tests/main/snap-cli-no-managers/task.yaml

```diff
@@ -0,0 +1,15 @@
+summary: Check that snap binary does not inadvertently import snapstate.
+description: |
+  This test checks that snap binary does not import snapstate when built
+  with -tags nomanagers.
+
+# XXX: enable on all systems once nomanagers tag is applied during packaging.
+systems: [ubuntu-1*, ubuntu-2*]
+
+execute: |
+  # XXX: just check /usr/bin/snap when packaging handles 'nomanagers' tag.
+  go build -tags nomanagers -o snap-test github.com/snapcore/snapd/cmd/snap
+  if strings snap-test | MATCH "overlord/snapstate"; then
```

Comment:
Maybe we don't need tags at all if we whitelist overlord/auth and overlord/state. We could also try something like this:
```
oimports="$(go list -f '{{range .Deps}}{{ . }}{{"\n"}}{{end}}' ./cmd/snap | grep 'github.com/snapcore/snapd/overlord/' | grep -Ev overlord/state | grep -Ev overlord/auth)"
if ! test -z "$oimports"; then
    echo "unexpected imports from overlord"
    echo "$oimports"
    exit 1
fi
```
You don't need a spread test anymore, it's enough to make it part of run-checks.

Reply:
[10:20:28]  <pstolowski> mborzecki: what do you mean by whitelist and not using tags?
[10:23:08]  <mborzecki> pstolowski: instead of building snap command with -tags nomanagers, you could check which overlord packages are imported, we know that overlord/state and overlord/auth are needed (for debug state and macaroons iirc) so those are the whitelist, everything else should throw an error
[10:23:51]  <pstolowski> mborzecki: but how? we need to import configcore, it just needs conditional build to exclude certain parts
[10:24:24]  <pstolowski> mborzecki: this dependency is introduced in my other PR
[10:24:31]  <mborzecki> aaah ok
[10:24:52]  <pstolowski> mborzecki: for that reason the simple dependency check will not work either
[10:25:02]  <mborzecki> pstolowski: which is the other pr?
[10:25:14]  <pstolowski> mborzecki: the spread test will be updated to check if packaging was done right (in followup(s))
[10:25:58]  <pstolowski> mborzecki: #8533. the conditional build will fix selinux denials there